src/ca/mcgill/cs/jetuml/diagram/ControlFlow.java

```diff
@@ -243,4 +246,124 @@ public boolean isInConstructorCall(Node pNode)
 		}
 		return false;	
 	}
+	
+	/*
+	 * Returns true if pNode is the ImplicitParameterNode that gets created in constructor call
+	 */
+	private boolean isImplicitParameterNodeInConstructorCall(Node pNode)
+	{
+		assert pNode != null;
+		return pNode.getClass() == ImplicitParameterNode.class && pNode.getChildren().size()>0 && isInConstructorCall(getFirstChild(pNode));
+	}
+	
+	private Node getFirstChild(Node pNode)
+	{
+		assert pNode.getChildren().size()>0;
+		return pNode.getChildren().get(0);
+	}
+	
+	private Optional<Edge> getConstructorEdge(Node pNode)
+	{
+		assert pNode != null && isInConstructorCall(pNode);
+		for(Edge edge : aDiagram.edges())
+		{
+			if ( pNode.getClass() == CallNode.class && edge.getEnd() == pNode && edge.getClass() == ConstructorEdge.class )
+			{
+				return Optional.of(edge);
+			}
+		}
+		return Optional.empty();	
+	}
+
+	/**
+	 * @param pEdge the Edge to get downstream elements from 
+	 * @return the downstream DiagramElements of pEdge

```

Comment:
Please add "excluding pEdge"

Reply:
The `pEdge` will be included because the code will call this method recursively. I added the "excluding pNode" in the `getNodeDownStreams` method.